src/libcore/ops/drop.rs

```diff
@@ -34,52 +34,97 @@
 /// }
 /// ```
 ///
-/// ## Dropping is done recursively
+/// Here, `main` will print `Dropping!`. In other words, the compiler generates code that
+/// kind of looks like this:
+///
+/// ```rust,compile_fail,E0040
+/// # struct HasDrop;
 ///
-/// When `outer` goes out of scope, the `drop` method will be called first for
-/// `Outer`, then for `Inner`. Therefore, `main` prints `Dropping Outer!` and
-/// then `Dropping Inner!`.
+/// # impl Drop for HasDrop {
+/// #     fn drop(&mut self) {
+/// #         println!("Dropping!");
+/// #     }
+/// # }
+/// fn main() {
+///     let _x = HasDrop;
 ///
+///     let mut x = _x;
+///     Drop::drop(&mut x);
+/// }
 /// ```
-/// struct Inner;
-/// struct Outer(Inner);
 ///
-/// impl Drop for Inner {
+/// Our custom implementation of `Drop` will be called at the end of `main`.
```

Comment:
A little misleading here, it seems as if there's a default implementation of `Drop`, which is not true.